waffle-jest/src/matchers/bigNumber.ts

```diff
@@ -0,0 +1,74 @@
+import {BigNumber} from 'ethers';
+import {Numberish} from '../types';
+
+// NOTE: Jest does not currently support overriding matchers while calling
+// original implementation, therefore we have to name our matchers something
+// different: https://github.com/facebook/jest/issues/6243
+
+export const bigNumberMatchers = {
+  toEqBN(received: Numberish, value: Numberish) {
+    const pass = BigNumber.from(received).eq(value);
+    return pass
```

Comment:
This pattern can be simplified and condensed into:
```ts
return {
  pass,
  message: () => pass
    ? `Expected "${received}" NOT to be greater than ${value}`
    : `Expected "${received}" to be greater than ${value}`
};
```

Reply:
I agree that's more concise, but I'd argue that it's a lot less readable because you are mixing a conditional into an object where one property should correspond to the other.

Reply:
More than fine for experimental merge